### PR TITLE
Add more failures rules to fail fast a Quarkus application

### PR DESCRIPTION
**Pull request by Sgitario**
Failure rules:
- Fail to start on DEV mode
- Fail to start when there is a missing or a wrong property

Fix https://github.com/quarkus-qe/quarkus-test-suite/issues/117

### CODE REVIEW

**Comment by Sgitario** [comment]
@rsvoboda I did check that with these changes, the framework does not wait up to 5 min for a Quarkus application that has failed.

File: quarkus-test-core/src/main/java/io/quarkus/test/services/quarkus/QuarkusManagedResource.java [hunk 1]
```diff
@@ -9,7 +12,9 @@
 public abstract class QuarkusManagedResource implements ManagedResource {
 
     private static final String EXPECTED_OUTPUT_FROM_SUCCESSFULLY_STARTED = "features";
-    private static final String UNRECOVERABLE_ERROR = "Failed to start application";
+    private static final List<String> ERRORS = Arrays.asList("Failed to start application",
```
Comment (Sgitario) [inline quarkus-test-core/src/main/java/io/quarkus/test/services/quarkus/QuarkusManagedResource.java:15]: In the future, this must be placed in an external resource to ease the maintenance.

**Comment by Sgitario** [comment]
run tests

**Review by rsvoboda — APPROVED** [review]

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Pull request merged**